src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -4348,7 +4348,8 @@ public void visitSelect(JCFieldAccess tree) {
                 if (env.info.isSelfCall &&
                         ((sym.name == names._this &&
                         site.tsym == env.enclClass.sym) ||
-                        sym.name == names._super && env.info.constructorArgs)) {
+                        sym.name == names._super && env.info.constructorArgs &&
```

Comment:
Maybe for later - let's refactor this to:
```
if (sym.name == names._this) {
   if (site.tsym == env.enclClass.sym) {
      // error
   }
} else if (sym.name == names._super && env.info.constructorArgs) {
   if (sitesym.isInterface() || site.tsym == env.enclClass.sym) {
      // error
   }
}
```

The `isSelfCall` should imply that `env.info.constructorArgs` is true, probably no need to check twice.